src/modules/socket.ts

```diff
@@ -2,147 +2,225 @@ import { io, Socket } from "socket.io-client";
+                        if (Math.abs(this.video.currentTime - message.currentTime) > 0.5 ) {
+                            const time1 = message.currentTime * 1000;
+                            const seekCode1 = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                                const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                                player.seek(${time1});`;
+                            document.documentElement.setAttribute(
+                                "onreset",
+                                seekCode1
+                            );
+                            document.documentElement.dispatchEvent(
+                                new CustomEvent("reset")
+                            );
+                        }
 
-  sendMessage = (message: string) => {
-    this.checkForErrors();
-    this.socket.emit(SocketEventType.MESSAGE, message, (response: string) => {
-      console.log("Response: " + response);
-    });
-  };
+                        console.log(this.video);
+                        const playCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                        const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                        player.play();`;
+                        document.documentElement.setAttribute(
+                            "onreset",
+                            playCode
+                        );
+                        document.documentElement.dispatchEvent(
+                            new CustomEvent("reset")
+                        );
+                        console.log("Video is played!");
+                        break;
+                    case "pause":
+                        console.log(this.video);
+                        const pauseCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                        const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
+                        player.pause();`;
+                        document.documentElement.setAttribute(
+                            "onreset",
+                            pauseCode
+                        );
+                        document.documentElement.dispatchEvent(
+                            new CustomEvent("reset")
+                        );
+                        console.log("Video is paused!");
+                        break;
+                    case "seeked":
+                        console.log("seeeek", this.seek);
+                        const time = message.currentTime * 1000;
+                        const seekCode = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
+                            const player = videoPlayer.getVideoPlayerBySessionId(videoPlayer.getAllPlayerSessionIds()[0]);
```

Comment:
This is the most important part of the change, when we receive the event, we use a netflix API call to control the video, if normal HTML function would be used, then the player would break.